### PR TITLE
FI-3378 Fix US Core importing

### PR DESCRIPTION
**Pull request by Shaumik-Ashraf**
# Summary

There was a bug reported where we could not launch `bundle exec inferno console` with local US Core tests given the way they were imported in PDex. This PR fixes the bug.

# Testing Guidance

1. Checkout this branch
2. `bundle exec rspec`
3. Launch services and test kit and check that it works like main branch
4. Add the line `gem 'us_core_test_kit', path: '../us-core-test-kit'` to Gemfile and run `bundle exec inferno console` and confirm that it works

### CODE REVIEW

File: lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb [hunk 1]
```diff
@@ -163,15 +165,27 @@ class PDexPayerServerSuite < Inferno::TestSuite
...
           
           group from: :pdex_explanation_of_benefit_group
 
           # Import all US Core v3.1.1 groups without the Suite
-          Dir.glob(File.join($LOAD_PATH.find { |x| x.match? "us_core_test_kit" }, 'us_core_test_kit/generated/v3.1.1/*_group.rb')).each do |test_group_path|
-            require_relative test_group_path
+          USCoreTestKit::USCoreV311::USCoreTestSuite.groups[1].groups.each do |group|
+            test_group = group.ancestors[1]
+
+            next if test_group.optional?
```
Comment (Jammjammjamm) [inline lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb:176]: Why are optional groups being excluded now?

File: lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb [hunk 2]
```diff
@@ -163,15 +165,27 @@ class PDexPayerServerSuite < Inferno::TestSuite
...
+
+            id = test_group.id
+
+            group_config = {}
+            if test_group.respond_to?(:metadata) &&
```
Comment (Jammjammjamm) [inline lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb:181]: Why is this code included?

File: lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb [hunk 3]
```diff
@@ -163,15 +165,27 @@ class PDexPayerServerSuite < Inferno::TestSuite
...
+               test_group.metadata.delayed? &&
+               !test_group.metadata.searchable_delayed_resource?
+              test_group.children.reject! { |child| child.include? USCoreTestKit::SearchTest }
+              group_config[:options] = { read_all_resources: true }
+            end
 
-            group from: "us_core_v311_#{File.basename(test_group_path).gsub('_group.rb','')}".to_sym
+            group(from: id, exclude_optional: true, config: group_config)
```
Comment (Jammjammjamm) [inline lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb:188]: Why are optional tests being excluded?

Comment (Shaumik-Ashraf) [reply lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb]: Fixed to include them all now

File: lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb (hunk 2 above)
Comment (Shaumik-Ashraf) [reply lib/davinci_pdex_test_kit/pdex_payer_server_suite.rb]: removed

**Review by Jammjammjamm — APPROVED** [review]

**Pull request merged**